lib/rubocop/cop/style/blocks.rb

```diff
@@ -27,18 +25,35 @@ def on_send(node)
         def on_block(node)
           return if ignored_node?(node)
 
-          block_length = Util.block_length(node)
-          block_begin = node.loc.begin.source
-
-          if block_length > 0 && block_begin == '{'
-            add_offense(node, :begin, MULTI_LINE_MSG)
-          elsif block_length == 0 && block_begin != '{'
-            add_offense(node, :begin, SINGLE_LINE_MSG)
+          if proper_block_style?(node)
+            correct_style_detected
+          else
+            add_offense(node, :begin) { opposite_style_detected }
           end
         end
 
         private
 
+        def message(node)
+          block_begin = node.loc.begin.source
+          block_length = Util.block_length(node)
+
+          case style
+          when :weirich
+            if block_begin == '{'
+              'Prefer do...end over {...} for procedural blocks.'
```

Comment:
Use backticks around `do...end` and `{...}` (this will colorize them in the console).